playbookutils_connector.py

```diff
@@ -0,0 +1,619 @@
+
+        # You should process the error returned in the json
+        message = f'Error from server. Status Code: {r.status_code} Data from server: {r.text.replace(u"{", "{{").replace(u"}", "}}")}'
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _process_response(self, r, action_result):
+        # store the r_text in debug data, it will get dumped in the logs if the action fails
+        if hasattr(action_result, 'add_debug_data'):
+            action_result.add_debug_data({'r_status_code': r.status_code})
+            action_result.add_debug_data({'r_text': r.text})
+            action_result.add_debug_data({'r_headers': r.headers})
+
+        # Process each 'Content-Type' of response separately
+
+        # Process a json response
+        if 'json' in r.headers.get('Content-Type', ''):
+            return self._process_json_response(r, action_result)
+
+        # Process an HTML response, Do this no matter what the api talks.
+        # There is a high chance of a PROXY in between phantom and the rest of
+        # world, in case of errors, PROXY's return HTML, this function parses
+        # the error and adds it to the action_result.
+        if 'html' in r.headers.get('Content-Type', ''):
+            return self._process_html_response(r, action_result)
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_response(r, action_result)
+
+        # everything else is actually an error at this point
+        message = (
+            f"Can't process response from server. Status Code: {r.status_code} "
+            f"Data from server: {r.text.replace(u'{', '{{').replace(u'}', '}}')}"
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, url, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        config = self.get_config()
+
+        resp_json = None
+
+        try:
+            request_func = getattr(ph_rules.requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, f'Invalid method: {method}'),
+                resp_json
+            )
+
+        try:
+            r = request_func(
+                url,
+                verify=config.get('verify_server_cert', False),
+                timeout=30,
+                **kwargs
+            )
+        except Exception as e:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, f'Error Connecting to server. Details: {e}'),
+                resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _determine_pb_run_id(self, action_result):
+        """ Figure out what the current playbook run id is by using the app_run_id, if it is running in a playbook.
```

Comment:
The default PB id logic should be mentioned to the user via README or param description as well.

Reply:
Hello @phantom-bneises,
Could you please look into this comment?

Reply:
Extended the method documentation a little further here.